backend/src/bundles/chat/chat.controller.ts

```diff
@@ -0,0 +1,131 @@
+import { type FastifySessionObject } from '@fastify/session';
+
+import {
+    type ApiHandlerOptions,
+    type ApiHandlerResponse,
+    BaseController,
+} from '~/common/controller/controller.js';
+import { ApiPath } from '~/common/enums/enums.js';
+import { HttpCode, HTTPMethod } from '~/common/http/http.js';
+import { type Logger } from '~/common/logger/logger.js';
+
+import { MAX_TOKEN } from './libs/constants/max-token.constant.js';
+import { ChatPath, OpenAIRole } from './libs/enums/enums.js';
+import { type GenerateTextRequestDto } from './libs/types/types.js';
+import { textGenerationValidationSchema } from './libs/validation-schemas/validation-schemas.js';
+import { type OpenAIService } from './open-ai.service.js';
+
+class ChatController extends BaseController {
+    private openAIService: OpenAIService;
+
+    public constructor(logger: Logger, openAIService: OpenAIService) {
+        super(logger, ApiPath.CHAT);
+
+        this.openAIService = openAIService;
+
+        this.addRoute({
+            path: ChatPath.ROOT,
+            method: HTTPMethod.PUT,
+            validation: {
+                body: textGenerationValidationSchema,
+            },
+            handler: (options) =>
+                this.generateChatAnswer(
+                    options as ApiHandlerOptions<{
+                        body: GenerateTextRequestDto;
+                        session: FastifySessionObject;
+                    }>,
+                ),
+        });
+
+        this.addRoute({
+            path: ChatPath.ROOT,
+            method: HTTPMethod.DELETE,
+            handler: (options) =>
+                this.clearChat(
+                    options as ApiHandlerOptions<{
+                        session: FastifySessionObject;
+                    }>,
+                ),
+        });
+
+        this.addRoute({
+            path: ChatPath.ROOT,
+            method: HTTPMethod.POST,
```

Comment:
Maybe I don't understand something, but why are you deleting the session with post and not delete method? It seems to me a little counterintuitive...